app/main.py

```diff
@@ -1 +1,61 @@
-#  write your code here
+class Animal:
+
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> any:
+        if self.is_hungry is True:
+            self.is_hungry = not self.is_hungry
+            print(f"Eating {self.appetite} food points...")
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+
+    def __init__(
+            self,
+            name: str,
+            appetite: int = 3,
+            is_hungry: bool = True
+    ) -> None:
+        super().__init__(name, is_hungry)
+        self.appetite = appetite
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+
+    def __init__(
+            self,
+            name: str,
+            appetite: int = 7,
+            is_hungry: bool = True) -> None:
+        super().__init__(name, is_hungry)
+        self.appetite = appetite
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list) -> int:
+    result = []
+    for animal in animals:
+        if isinstance(animal, Animal) and animal.is_hungry is True:
+            animal.feed()
+            result.append(animal.appetite)
```

Comment:
This operation can be simplified into a one-liner using generator expression with `sum` for improved code conciseness.

Reply:
i tryed, but i have no idea, how in generator execute 2 functions (animal.(feed() and generate list with animal.appetite), explain to me, please.

Reply:
Instead of initializing an empty list to collect results, we use a generator expression inside sum.

The generator expression iterates over each animal in the animals list.

If the conditions are met, the animal.feed() method is called directly within the generator expression.

The generator expression itself produces a sequence of appetite values returned by animal.feed().

sum then calculates the total sum of these values and returns it.

Reply:
ok, maybe i find solution)))

Reply:
aaaa, self.feed return self.appetite)))) I forgot about this. maybe my last request not so correct)